operator/internal/controller/podgangset/register.go

```diff
@@ -37,7 +37,7 @@ func (r *Reconciler) RegisterWithManager(mgr manager.Manager) error {
 	return builder.ControllerManagedBy(mgr).
 		Named(controllerName).
 		WithOptions(controller.Options{
-			MaxConcurrentReconciles: *r.config.ConcurrentSyncs,
+			MaxConcurrentReconciles: *r.config,
 		}).
```

Comment:
`r.config` can not be indirected here, and is not the right type either.
```suggestion
			MaxConcurrentReconciles: *r.config.ConcurrentSyncs,
```